sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/LatencyRecordingHttpRequestInitializer.java

```diff
@@ -31,6 +31,7 @@
 
 /** HttpRequestInitializer for recording request to response latency of Http-based API calls. */
 public class LatencyRecordingHttpRequestInitializer implements HttpRequestInitializer {
+  // TODO: Import the URN from MonitoringInfoConstants.Urns when it's available
   public static final String HISTOGRAM_URN = " beam:metric:io:api_request_latencies:v1";
```

Comment:
Since you are instantiing the latency monitoring info, please add a MonitoringInfoSpec for it, and pull it its urn as the constant here
https://github.com/apache/beam/blob/eca935bc36036dd11c5f2bd20052b08049375bfc/model/pipeline/src/main/proto/metrics.proto#L69

This would be ideal, as we should not instantiate metrics for URNs without a MonitoringInfoSpec.


It should be very similar to the API_REQUEST_COUNT, without STATUS label, a different URN, and a histogram type

i.e. you can use

    API_REQUEST_LATENCIES = 20 [(monitoring_info_spec) = {
      urn: "beam:metric:io:api_request_latencies:v1",
      type: "beam:metrics:histogram_int64:v1",
      required_labels: [
        "SERVICE",
        "METHOD",
        "RESOURCE",
        "PTRANSFORM"
      ],
      annotations: [
         {
            key: "description",
            value: "Histogram counts for request latencies made to an IO’s service APIs to batch read or write elements."
        }, 
        {
          key: "units",
          value: "Milliseconds"
        },
        {
          key: "process_metric",  // Should be reported as a process metric
                                  // instead of a bundle metric
          value: "true"
        }
      ]
    }];

Reply:
Done.